dev-reqs.txt

```diff
@@ -1,7 +1,10 @@
+beautifulsoup4==4.6.0
+lxml==3.8.0
 codeclimate-test-reporter
 coveralls
 flake8
 flask_cors
+lxml==3.8.0
```

Comment:
once should be enough

Reply:
actually i can't find where you are using lxml and beautifulsoup

Reply:
The are required by `pandas.read_html()`, which I use in `connector_tests` to make it easy to set up test data as a Markdown table and then push it into whatever Datasource is being tested.

Reply:
Then they should be added to setup.py or we are going to miss them in production :)

Reply:
https://pandas.pydata.org/pandas-docs/stable/io.html#io-html-gotchas explains that `read_html` uses bs4 as a wrapper for lxml. We could use `html5lib` instead of `lxml`.

Reply:
The original comment was because you added lxml twice not because you added two different libraries :)

Reply:
> The original comment was because you added lxml twice

Thanks - I've fixed it now.

Reply:
> Then they should be added to setup.py or we are going to miss them in production

We only *need* them for `connector_tests`. I am assuming that we don't need to be able to run the tests in production (hence `dev-reqs.txt`). You would only need them in production if you wanted to be able to use an html-format PandasDatasource. But there are many possible extra libraries that we might need to support the full range of `read_xxx` Pandas functionality - hdf5, xlrd, probably something for Stata. I was assuming that we wouldn't put all the possible dependencies into Superset, and we would expect the individual implementations to install the correct libraries for the formats they want/need to support.

If it is preferable that Superset supports the full range of Pandas `read_xxx` out of the box then I could update `setup.py` to include them and add tests. But that only makes sense if PandasConnector is going to be in core rather than in a `contrib` directory.

Reply:
If html is a format supported, wouldn't parse_html used then?

Reply:
@xrmx - you are correct - we are specifically supporting html in `FORMATS` so I need to add those dependencies to `setup.py` or whatever requirements file we ship with PandasConnector if it lives outside core.

I wonder if there are any other formats we should support by default? JSON should work by default. Excel might be useful for many people and would need `xlrd`? Given the normal Superset audience do we want to support HDF5 and/or Feather? What about SAS or Stata?

Reply:
In my opinion we should keep the pandas data source only for the simple cases: excel, csv, html and json. For more complex scenarios i think an apache arrow backend as suggested by @mistercrunch is the way to go.

Reply:
I agree that Arrow is the future. http://wesmckinney.com/blog/apache-arrow-pandas-internals/ is a good read.

Reply:
I have added the necessary packages to `setup.py` so that `PandasDatasource` works for csv, html, json and Excel.

@xrmx do you think it would be a good idea to allow `PandasDatasource.FORMATS` to be overridden from settings, much as we allow `ADDITIONAL_MODULE_DS_MAP` to add extra Connectors? Then a specific installation could, for example, install `pytables` and then set:

```python
ADDITIONAL_PANDAS_FORMATS = [('hdf', 'HDF5')]
```

Another approach would be to attempt to dynamically set the list of allowable formats based on whether we can import the required dependencies.

Reply:
Uhm, we probably don't want to tie the supported formats to the db so maybe it's better to just have a varchar in the db and limit the values from a select built from a config key so each user can support whatever formats they want.